nullaway/src/main/java/com/uber/nullaway/fixserialization/Serializer.java

```diff
@@ -195,4 +203,116 @@ public static Path pathToSourceFileFromURI(@Nullable URI uri) {
       return path;
     }
   }
+
+  /**
+   * Serializes the given {@link Symbol} to a string.
+   *
+   * @param symbol The symbol to serialize.
+   * @return The serialized symbol.
+   */
+  public static String serializeSymbol(@Nullable Symbol symbol) {
+    if (symbol == null) {
+      return "null";
+    }
+    switch (symbol.getKind()) {
+      case FIELD:
+      case PARAMETER:
+        return symbol.name.toString();
+      case METHOD:
+      case CONSTRUCTOR:
+        return serializeMethodSignature((Symbol.MethodSymbol) symbol);
+      default:
+        return symbol.flatName().toString();
+    }
+  }
+
+  /**
+   * Serializes the signature of the given {@link Symbol.MethodSymbol} to a string.
+   *
+   * @param methodSymbol The method symbol to serialize.
+   * @return The serialized method symbol.
+   */
+  private static String serializeMethodSignature(@Nullable Symbol.MethodSymbol methodSymbol) {
+    if (methodSymbol == null) {
+      return "null";
+    }
+    StringBuilder sb = new StringBuilder();
+    if (methodSymbol.isConstructor()) {
+      // For constructors, we use the name of the enclosing class.
+      Name name = methodSymbol.owner.enclClass().getSimpleName();
+      if (name.isEmpty()) {
+        name = methodSymbol.owner.enclClass().getSuperclass().asElement().getSimpleName();
+      }
```

Comment:
can you add a comment that this is for anonymous classes?  Also, store `methodSymbol.owner.enclClass()` in a local variable so we don't write it twice

Reply:
bc4c6e0dca9d09d66028a5dc8daf40fdeef40562